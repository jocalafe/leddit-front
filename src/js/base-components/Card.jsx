import React from 'react';
import Card from 'material-ui/Card';
import PropTypes from 'prop-types';

const LedditCard = ({ children,
  style,
  containerStyle,
  expandable = false,
  expanded,
  initiallyExpanded = false,
  onExpandChange,
  showExpandableButton}) => (
    <div>
      <Card style={style}
        containerStyle={containerStyle}
        expandable={expandable}
        expanded={expanded}
        initiallyExpanded={initiallyExpanded}
        onExpandChange={onExpandChange}
        showExpandableButton={showExpandableButton}>
        {children}
      </Card>
    </div>
);

LedditCard.propTypes = {
  children: PropTypes.func,
  style: PropTypes.string,
  containerStyle: PropTypes.string,
  expandable: PropTypes.bool,
  expanded: PropTypes.bool,
  initiallyExpanded: PropTypes.bool,
  onExpandChange: PropTypes.func,
  showExpandableButton: PropTypes.func,
};

export default LedditCard;
