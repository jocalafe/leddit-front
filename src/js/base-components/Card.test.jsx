import React from 'react';
import renderer from 'react-test-renderer';

import Card from './Card';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';

jest.mock('react-dom');

describe('Card', () => {
  it('renders', () => {
    const tree = renderer.create(
      <MuiThemeProvider>
        <Card />
      </MuiThemeProvider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
