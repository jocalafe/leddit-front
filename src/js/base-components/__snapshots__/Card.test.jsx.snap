// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders 1`] = `
<div>
  <div
    showExpandableButton={undefined}
    style={
      Object {
        "WebkitTapHighlightColor": "rgba(0,0,0,0)",
        "backgroundColor": "#ffffff",
        "borderRadius": 2,
        "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
               0 1px 4px rgba(0, 0, 0, 0.12)",
        "boxSizing": "border-box",
        "color": "rgba(0, 0, 0, 0.87)",
        "fontFamily": "Roboto, sans-serif",
        "muiPrepared": true,
        "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
        "zIndex": 1,
      }
    }
  >
    <div
      style={
        Object {
          "paddingBottom": 0,
        }
      }
    />
  </div>
</div>
`;
